assignment-01-cat/Cat.cpp

```diff
@@ -1,15 +1,69 @@
 #include "Cat.h"
+#include <cstring>
 
-Cat::Cat(const char* _name) {
+Cat::Cat(const char* _name)
+{
     // TODO: this->name에 동적할당을 하고 _name을 복사해야 합니다.
+    this->name = new char[strlen(_name)+1];
+    strcpy(name, _name);
     // TODO: this->age는 -1으로 초기화해야 합니다.
+    this->age = -1;
     // TODO: this->sex는 false로 초기화해야 합니다.
+    this->sex = false;
 }
 
-Cat::Cat(const char* _name, int _age) {
+Cat::Cat(const char* _name, int _age)
+{
     // TODO: this->name에 동적할당을 하고 _name을 복사해야 합니다.
-    // TODO: this->age에 _age로 초기화해야 합니다.
+    this->name = new char[strlen(_name)+1];
+    strcpy(name, _name);
+    // TODO: this->age에 _age로 초기화야 합니다.
+    this->age = _age;
     // TODO: this->sex는 false로 초기화해야 합니다.
+    this->sex = false;
 }
 
-// TODO: 나머지 모두 구현하세요.
+Cat::Cat(const char* _name, int _age, bool _sex)
+{
+    this->name = new char[strlen(_name)+1];
+    strcpy(name, _name);
+    this->age = age;
+    this->sex = false;
+}
```

Comment:
```suggestion
Cat::Cat(const char* _name, int _age, bool _sex): age(_age), sex(_sex)
{
    this->name = new char[strlen(_name) + 1];
    strcpy(name, _name);
}
```
단순 대입 대신 초기화 리스트를 써보는 게 어떨까요?
그런데 현재 코드는 `_sex` 매개변수를 활용하지 않고 있네요 ㅠㅠ